packages/headless/src/controllers/core/search-parameter-manager/headless-core-search-parameter-manager.test.ts

```diff
@@ -215,30 +217,54 @@ describe('search parameter manager', () => {
       expect(manager.state.parameters.df).toEqual({created: [selected]});
     });
 
-    it('when there are no date facets with selected values, the #df parameter is not included', () => {
+    it('is not included when there are no date facets with selected values', () => {
       engine.state.dateFacetSet = {created: buildMockDateFacetSlice()};
-      expect('df' in manager.state.parameters).toBe(false);
+      expect(manager.state.parameters).not.toContain('df');
+    });
+  });
+
+  describe('#state.parameters.af', () => {
+    it('only includes selected values when a facet has some', () => {
+      const selected = buildMockFacetValue({
+        value: 'a',
+        state: 'selected',
+      });
+      const idle = buildMockFacetValue({value: 'b', state: 'idle'});
+
+      const currentValues = [selected, idle];
+      engine.state.automaticFacetSet.facets = {
+        author: buildMockAutomaticFacetResponse({values: currentValues}),
+      };
+
+      expect(manager.state.parameters.af).toEqual({author: ['a']});
+    });
+
+    it('is not included when there are no facets with selected values', () => {
+      engine.state.automaticFacetSet.facets = {
+        author: buildMockAutomaticFacetResponse(),
+      };
+      expect(manager.state.parameters).not.toContain('af');
     });
   });
 
   describe('#state.parameters.sortCriteria', () => {
-    it('when the parameter does not equal the default value, it is included', () => {
+    it('is included when the parameter does not equal the default value, ', () => {
       engine.state.sortCriteria = 'qre';
       expect(manager.state.parameters.sortCriteria).toBe('qre');
     });
 
-    it('when the parameter is equal to the default value, it is not included', () => {
-      expect('sortCriteria' in manager.state.parameters).toBe(false);
+    it('is not included when the parameter is equal to the default value', () => {
+      expect(manager.state.parameters).not.toContain('sortCriteria');
     });
 
-    it('when the parameter is undefined, it is not included', () => {
+    it('is not included when the parameter is undefined', () => {
       engine.state.sortCriteria = undefined as never;
-      expect('sortCriteria' in manager.state.parameters).toBe(false);
+      expect(manager.state.parameters).not.toContain('sortCriteria');
     });
   });
 
-  it(`given a certain initial state,
-  it is possible to access every relevant search parameter using #state.parameters`, () => {
+  it(`is possible to access every relevant search parameter using #state.parameters
```

Comment:
is there string interpolation happening here?
also nit: remove `,` at end

Reply:
The string interpolation is needed for the line break. 

I prefer like this : 
```
it(`is possible to access every relevant search parameter using #state.parameters
   given a certain initial state`, () => {
```
Than this:
```
it('is possible to access every relevant search parameter using #state.parameters given a certain initial state', () => {
```

Reply:
That is weird. Are there any other instances where `` are used to break string across lines?

Reply:
Yes there must be other instances in our test files where we do this. It was done this way before I made my changes. If you really dont like it, I will remove them. I just think its more readable this way.

Reply:
I removed them all

Reply:
Thanks, I asked because it felt strange (not incorrect) and wanted to make sure if the pattern was used in the codebase already. Although the new line might improve readability, when using string interpolation the reader would expect to find a variable inside if not it might lead to some pause/wonder - which would reduce its readability.